docs/designs/012-health-events-exporter.md

```diff
@@ -0,0 +1,546 @@
+        "generatedTimestamp": healthEvent.GeneratedTimestamp.AsTime().Format(time.RFC3339Nano),
+        "nodeName":          healthEvent.NodeName,
+        "quarantineOverrides": map[string]interface{}{
+            "force": healthEvent.QuarantineOverrides.Force,
+            "skip":  healthEvent.QuarantineOverrides.Skip,
+        },
+        "drainOverrides": map[string]interface{}{
+            "force": healthEvent.DrainOverrides.Force,
+            "skip":  healthEvent.DrainOverrides.Skip,
+        },
+    },
+}
+
+// transformEntities converts protobuf entities to JSON-friendly format
+func transformEntities(entities []*pb.Entity) []map[string]string {
+    result := make([]map[string]string, len(entities))
+    for i, entity := range entities {
+        result[i] = map[string]string{
+            "entityType":  entity.EntityType,
+            "entityValue": entity.EntityValue,
+        }
+    }
+    return result
+}
+```
+
+
+### 3. Event Stream Pipeline
+
+The exporter watches all insert operations on the health events collection:
+
+```go
+// BuildHealthEventsExportPipeline creates a pipeline for health events
+// Works with any datastore provider (MongoDB, PostgreSQL, etc.)
+func BuildHealthEventsExportPipeline() interface{} {
+    return EventFilter{
+        OperationType: "insert",  // Watch all new health event inserts
+    }
+}
+
+// Provider-specific implementation:
+// MongoDB: db.healthevents.watch([{$match: {operationType: "insert"}}])
+// PostgreSQL: LISTEN health_events WHERE operation = 'INSERT'
+// Kafka: Consume from health-events topic
+```
+
+### 4. Sink Implementations
+
+#### HTTP Sink
+
+```go
+type HTTPPublisher interface {
+    // Publish single event
+    Publish(ctx context.Context, event CloudEvent) error
+    
+    // Close publisher and flush pending events
+    Close(ctx context.Context) error
+}
+
+type HTTPPublisherConfig struct {
+    // Sink configuration
+    Endpoint            string        // HTTP endpoint URL
+    MaxRetries          int           // Number of retries on failure
+    RetryBackoff        time.Duration // Initial backoff duration
+    Timeout             time.Duration // Request timeout
+    
+    // Authentication
+    OIDCTokenProvider   TokenProvider // OIDC token provider
+    
+    // HTTP client settings
+    MaxIdleConns        int
+    MaxIdleConnsPerHost int
+    IdleConnTimeout     time.Duration
+}
+```
+
+#### Authentication
+
+HTTP sink uses **OAuth2 Client Credentials Flow** (OIDC): client credentials configured via secrets, tokens auto-refreshed and cached, included in requests as `Authorization: Bearer <token>`.
+
+### 5. Event Processing Flow
+
+Pipeline: Receive → Transform → Publish → Update resume token → Repeat
+
+Errors: Transform/publish failures logged and skipped (with HTTP retry); resume token failures trigger restart to maintain consistency.
```

Comment:
In some cases, the sink may be temporarily unavailable say for like 30m/60m for a maintenance. How do we handle such cases? Can we have a large timeout say something like 6 hours?

Reply:
Added retries upto 30 min in code, rest will be on k8s restart